### DISCUSSION TITLE
View Size

### DISCUSSION THREAD

**Discussion by mutaboss** [gyscos/cursive#599] [Q&A]
I'm writing a maze generation program and want to have dialogs around it to customize what happens. What I would l like to do is be able to detect the size of the screen and generate a maze that will fit in that area, but I can't seem to find a reliable way to get the screen size from Cursive. Is that possible?

**Comment by gyscos** [comment]
Hi!

There are a few ways to achieve that:
* You can call `Cursive::screen_size()` to get the last used screen size. Note that:
  * This will only be available after the backend has been initialized, which usually happens in the call to `Cursive::run()`. This means that it's not available before running the event loop.
  * It will return the entire available space for the application. If you use margins or padding (for example if you add regular layers) you will have to subtract that yourself.
* You can get the available size for an individual view in its call to `View::layout`. So if you implement a `View` yourself (or if you use a `Canvas`), you can put the maze-building logic in the `layout` function and use the size there.

**Reply by mutaboss** [reply]
Ah, totally. I implemented a custom view and got it working. Sweet! Clearly it has been far too long since I have done any GUI programming, I'll need to rethink my program a bit, but I'm now moving forward again. Thanks for the help!
